### PR TITLE
[gluster] Optionally enable dump

### PR DESCRIPTION
**Pull request by psachin**
Currently a dump is collected every time sosreport is run. Dump is not
required by Gluster support team and also consumes space in
`/var/run/gluster/` unnecessarily.

This patch adds an option to enable dump when required using
```
-k gluster.dump
```

Command
```
sosreport -o gluster -k gluster.dump
```

Dump will is disabled by default.

(Few `pylint` changes are also included with this patch which are not
related to enhancement. Let me know if you want me to remove those.)

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

File: sos/plugins/gluster.py [hunk 1]
```diff
@@ -111,32 +113,31 @@ def setup(self):
...
-                                    "gluster volume heal %s info" % volname,
-                                    "gluster volume heal %s info split-brain"
-                                    % volname,
-                                    "gluster snapshot list %s" % volname,
-                                    "gluster volume quota %s list" % volname,
-                                    "gluster volume rebalance %s status"
-                                    % volname,
-                                    "gluster snapshot info %s" % volname,
-                                    "gluster snapshot status %s" % volname])
+                self.add_cmd_output(
```
Comment (TurboTurtle) [inline sos/plugins/gluster.py]: Minor styling correction here. Put the starting bracket in the same line as `add_cmd_output` and the closing bracket and paren should be on a new line with the same indention as the call.

E.G.:
```
self.add_cmd_output([
    ...list...
])
```

**Comment by psachin** [comment]
Hi @TurboTurtle 

I fixed the indentation to match with `add_copy_spec()` above and to also pass `flake8` test. Let me know if that's OK.

**Comment by TurboTurtle** [comment]
LGTM

**Comment by psachin** [comment]
@bmr-cymru Can you please review as well?

**Comment by bmr-cymru** [comment]
Merged, thanks!

**Pull request closed without merging**